### ISSUE TITLE
semihosting support?

### ISSUE THREAD

**Issue by smithbone** [renode/renode#681]
### Description

Create an option to dynamically enable semihosting support.  According to here it looks like renode has to be rebuilt? https://docs.platformio.org/en/stable/advanced/unit-testing/semihosting.html

`machine EnableSemihosting`

### Usage example

See the multitude of examples in the QEMU project

### Additional information

QEMU provides this support and it is very helpful for working with a MCU that is a supported architecture but doesn't have a working uart model or bringing up a new software platform on hardware that is supported but doesn't have a working uart driver yet.  

### Do you plan to address this issue and file a PR?
No.

**Comment by johannesgensheimer** [comment]
Agree! Does logging from the code work when debugging with GDB or do we also have to redirect to UART here?

**Comment by jwolski-main** [comment]
Recently we updated our semihosting handling, please check if it works for you. You can attach the handler by adding the following to your board's .repl file.
```
semihosting: CPU.SemihostingHandler @ cpu
uart: UART.SemihostingUart @ semihosting
```
After that semihosting should be enabled by default. 
Example: https://github.com/antmicro/renode/blob/master/platforms/cpus/a20.repl#L71-L73
For detailed insight look at [SemihostingHandler](https://github.com/renode/renode-infrastructure/blob/master/src/Emulator/Cores/Arm/SemihostingHandler.cs) class.

Please open a new issue if problem still arises.
